libgc/os_dep.c

```diff
@@ -26,7 +26,7 @@
 #     define __KERNEL__
 #     include <asm/signal.h>
 #     undef __KERNEL__
-#   else
+#   elif defined(__GLIBC__)
```

Comment:
For Alpine, perhaps we can detect and disable Boehm?

I am not crazy about adding libc-specific defines.